Google.PowerShell/Compute/GceForwardingRuleCmdlets.cs

```diff
@@ -0,0 +1,174 @@
+            public const string ByLocalName = "ByLocalName";
+            public const string ByGlobalName = "ByGlobalName";
+        }
+
+        /// <summary>
+        /// <para type="description">
+        /// The project the forwarding rules belong to. Defaults to the project in the Cloud SDK config.
+        /// </para>
+        /// </summary>
+        [Parameter(ParameterSetName = ParameterSetNames.OfProject)]
+        [Parameter(ParameterSetName = ParameterSetNames.OfRegion)]
+        [Parameter(ParameterSetName = ParameterSetNames.ByLocalName)]
+        [Parameter(ParameterSetName = ParameterSetNames.ByGlobalName)]
+        [ConfigPropertyName(CloudSdkSettings.CommonProperties.Project)]
+        public string Project { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// If set, will retrieve only global forwarding rules.
+        /// </para>
+        /// </summary>
+        [Parameter(ParameterSetName = ParameterSetNames.OfProject)]
+        [Parameter(ParameterSetName = ParameterSetNames.ByGlobalName, Mandatory = true)]
+        public SwitchParameter Global { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The region of the forwaring rule to get. Defaults to the region in the Cloud SDK config.
+        /// </para>
+        /// </summary>
+        [Parameter(ParameterSetName = ParameterSetNames.OfRegion, Mandatory = true)]
+        [Parameter(ParameterSetName = ParameterSetNames.ByLocalName)]
+        [ConfigPropertyName(CloudSdkSettings.CommonProperties.Region)]
+        public string Region { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The name of the forwarding rule to get.
+        /// </para>
+        /// </summary>
+        [Parameter(ParameterSetName = ParameterSetNames.ByLocalName, Mandatory = true, Position = 0)]
+        [Parameter(ParameterSetName = ParameterSetNames.ByGlobalName, Mandatory = true, Position = 0)]
+        public string Name { get; set; }
+
+        protected override void ProcessRecord()
+        {
+            switch (ParameterSetName)
+            {
+                case ParameterSetNames.OfProject:
+                    WriteObject(GetAllProjectForwardingRules(), true);
+                    break;
+                case ParameterSetNames.OfRegion:
+                    WriteObject(GetRegionForwardingRules(), true);
+                    break;
+                case ParameterSetNames.ByLocalName:
+                    WriteObject(Service.ForwardingRules.Get(Project, Region, Name).Execute());
+                    break;
+                case ParameterSetNames.ByGlobalName:
+                    WriteObject(Service.GlobalForwardingRules.Get(Project, Name).Execute());
+                    break;
+                default:
+                    throw UnknownParameterSetException;
+            }
+        }
+
+        private IEnumerable<ForwardingRule> GetRegionForwardingRules()
```

Comment:
This code is fine, but consider making it stateless. i.e. pass the project and region as parameters, so that the function does not depend on `this`. It's a good principle as it makes code easier to unit test, etc. But feel free to disregard in this situation.